crates/sargon-uniffi/src/profile/v100/factors/factor_source.rs

```diff
@@ -58,6 +58,16 @@ pub fn factor_source_name(factor_source: &FactorSource) -> String {
     factor_source.into_internal().name()
 }
 
+#[uniffi::export]
+pub fn factor_source_set_name(
```

Comment:
This is nice, but I think we can go a step further and also make the update in the Profile itself, so maybe this API should have the factor source id to be updated.

Or your intention is different?

Reply:
I was thinking of handling it the same way we handle Account name updates, but guess we can directly do altogether. Will update

Reply:
That is more legacy :), we should migrate how we rename accounts sometime to be handled in Sargon.

Reply:
actually I am not sure if it is better to have a function that updates the `name: String` of a given `factor_source_id: FactorSourceId`. Considering Hosts we will already have access to the actual `FactorSource` and that we already have a [function](https://github.com/radixdlt/sargon/blob/main/crates/sargon/src/system/sargon_os/sargon_os_factors.rs#L32) that updates a `FactorSource`, seems pointless to send the `id` just to have Sargon fetch it from the Profile.

Maybe instead the function should be like this
```Rust
pub async fn update_factor_source_name(&self, factor_source: FactorSource, name: String) -> Result<()>
```

Reply:
Sure that also works, the main idea is to have the Profile mutation in Sargon. You might want to return the update FactorSource as response of this function.